### PR TITLE
Gw v0.10.1

### PR DESCRIPTION
**Pull request by kcleal**

### CODE REVIEW

**Comment by kcleal** [comment]
Linker cannot find libs. How should these be defined in the build?
```
ld.lld: error: unable to find library -lpthread
[2918](https://github.com/termux/termux-packages/actions/runs/9841341897/job/27167721840?pr=20796#step:8:2919)
ld.lld: error: unable to find library -lX11
[2919](https://github.com/termux/termux-packages/actions/runs/9841341897/job/27167721840?pr=20796#step:8:2920)
ld.lld: error: unable to find library -lhts
[2920](https://github.com/termux/termux-packages/actions/runs/9841341897/job/27167721840?pr=20796#step:8:2921)
ld.lld: error: unable to find library -lfreetype
[2921](https://github.com/termux/termux-packages/actions/runs/9841341897/job/27167721840?pr=20796#step:8:2922)
ld.lld: error: unable to find library -luuid
[2922](https://github.com/termux/termux-packages/actions/runs/9841341897/job/27167721840?pr=20796#step:8:2923)
ld.lld: error: unable to find library -lcurl
[2923](https://github.com/termux/termux-packages/actions/runs/9841341897/job/27167721840?pr=20796#step:8:2924)
ld.lld: error: unable to find library -lglfw
[2924](https://github.com/termux/termux-packages/actions/runs/9841341897/job/27167721840?pr=20796#step:8:2925)
ld.lld: error: unable to find library -lfontconfig
```

**Comment by TomJo2000** [comment]
`gw` has `TERMUX_PKG_AUTO_UPDATE` set to `true`.
https://github.com/termux/termux-packages/blob/224857d7521ca9a2b606a875a64ffabfefc875ac/x11-packages/gw/build.sh#L1-L11
It would have auto-updated if there weren't build issues with the new version.
As such a simple version bump and updating the checksum is not going to be enough in this case.

<h1><em></em></h1> 

From the CI logs, it looks like it is unable to find (most of?) its build time dependencies.
And also requires `curl` as a dependency now.
```st
aarch64-linux-android-clang++  -fstack-protector-strong -Oz -Wall -std=c++17 -fno-common -fwrapv -fno-omit-frame-pointer -O3 -DNDEBUG -g -D LINUX -D __STDC_FORMAT_MACROS  ./src/drawing.o  ./src/glfw_keys.o  ./src/hts_funcs.o  ./src/main.o  ./src/menu.o  ./src/parser.o  ./src/plot_commands.o  ./src/plot_controls.o  ./src/plot_manager.o  ./src/segments.o  ./src/term_out.o  ./src/themes.o  ./src/utils.o  ./lib/libBigWig/bwRead.o  ./lib/libBigWig/bwValues.o  ./lib/libBigWig/bwWrite.o  ./lib/libBigWig/io.o -L./lib/skia/out/Release-arm64 -lskia -lm -ljpeg -lpng -lpthread -lX11 -lEGL -lGLESv2 -lhts -lfreetype -luuid -lz -lcurl -licu -ldl -lglfw -lsvg -lfontconfig -o gw
ld.lld: error: unable to find library -lpthread
ld.lld: error: unable to find library -lX11
ld.lld: error: unable to find library -lhts
ld.lld: error: unable to find library -lfreetype
ld.lld: error: unable to find library -luuid
ld.lld: error: unable to find library -lcurl
ld.lld: error: unable to find library -lglfw
ld.lld: error: unable to find library -lfontconfig
clang-17: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:125: gw] Error 1
Error: Process completed with exit code 2.
```

**Comment by kcleal** [comment]
Thanks @TomJo2000 
Some of those libs are in the build recipe already, some I think should be on the system already (X11, pthread), so not sure why the linker is failing?

**Comment by kcleal** [comment]
It looks like there should be something added to LDLIBS or similar, the linker can't find various libs, even though some of them are in the recipe? Does anyone know how to fix this? Thanks!

**Comment by truboxl** [comment]
Patch the makefiles or use LDFLAGS with `-L${TERMUX_PREFIX}/lib`

**Comment by truboxl** [comment]
https://github.com/kcleal/gw/blob/b4f4706ba94fb5b86fb5eca4d9fce68e99e0102e/Makefile#L55

Ah I fail to mention that LDFLAGS already have `-L${TERMUX_PREFIX}/lib` inside our build scripts.

It's just that the Makefile is clearing out the LDFLAGS, so you have remove that line by patching it.

**Review by TomJo2000 — APPROVED** [review]

File: x11-packages/gw/build.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ termux_step_pre_configure() {
 	sed -i \
 		-e '/\/usr\/local\/include/d' \
 		-e '/\/usr\/local\/lib/d' \
-		./Makefile
+		./Makefile && sed -i 's/LDFLAGS=/LDFLAGS+=/g' Makefile
```
Comment (thunder-coding) [inline x11-packages/gw/build.sh:23]: This can be merged into a single command

File: x11-packages/gw/build.sh [hunk 2]
```diff
@@ -55,7 +55,7 @@ termux_step_make() {
 	unzip -o "${TERMUX_PKG_CACHEDIR}/skia-${TERMUX_ARCH}.zip"
 	cd ../../
 
-	make -j "$TERMUX_PKG_MAKE_PROCESSES"
+	LDFLAGS="${LDFLAGS} -L${TERMUX_PREFIX}/lib" make -j "$TERMUX_PKG_MAKE_PROCESSES"
```
Comment (thunder-coding) [inline x11-packages/gw/build.sh:58]: Is this actually needed?

build-package.sh:642 initialises the `LDFLAGS` environment variable, which is called before `termux_step_pre_configure`

**Review by thunder-coding — CHANGES_REQUESTED** [review]

File: x11-packages/gw/build.sh (hunk 1 above)
Comment (thunder-coding) [reply x11-packages/gw/build.sh:23]: This comment has not been resolved

**Comment by TomJo2000** [comment]
The build error was caused by an overlooked array reference in `alist`'s build script.
Fixed it in https://github.com/termux/termux-packages/commit/aa97f6e3eed35907c0da2ce748dfdeefbe90d7cf.

I'm going to rebase this PR on top of that commit.

**Comment by TomJo2000** [comment]
And, we're all green.

File: x11-packages/gw/build.sh [hunk 3]
```diff
@@ -2,12 +2,12 @@ TERMUX_PKG_HOMEPAGE=https://github.com/kcleal/gw
 TERMUX_PKG_DESCRIPTION="A browser for genomic sequencing data (.bam/.cram format)"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="clealk@cardiff.ac.uk"
-TERMUX_PKG_VERSION=0.9.3
+TERMUX_PKG_VERSION=0.10.1
 TERMUX_PKG_REVISION=1
```
Comment (tqfx) [inline x11-packages/gw/build.sh:6]: Delete revision as it shouldn't be present if package was updated.
```suggestion
```

File: x11-packages/gw/build.sh (hunk 2 above)
Comment (tqfx) [reply x11-packages/gw/build.sh:58]:
```suggestion
	make -j "$TERMUX_PKG_MAKE_PROCESSES"
```

Comment (thunder-coding) [reply x11-packages/gw/build.sh:57]: Once this change is not needed, please undo this

**Review by thunder-coding — CHANGES_REQUESTED** [review]

**Comment by TomJo2000** [comment]
No idea what went wrong with the previous CI run, we're all green here now.

Package contents check out.
I think we have all concerns addressed?
I'd like to just get this PR done.

**Review by thunder-coding — APPROVED** [review]
lgtm

**Comment by TomJo2000** [comment]
Just gonna wait out the CI run, and then we can squash this mess.

**Comment by thunder-coding** [comment]
Also @kcleal, please consider squashing your commits after rebasing against the master branch. It makes the review process much easier for us. Also thanks for the PR

**Comment by TomJo2000** [comment]
I'm fine doing rebases and squashing myself.
I just generally prefer to leave the PR author in control of what exactly is in what commit, and what the commit message(s) are.

**Comment by TomJo2000** [comment]
Think that might be our first 3 author commit in the repo lol.

**Pull request merged**